### PR TITLE
fix(events): more thorough verification of events in sui checkpoints

### PR DESCRIPTION
**Pull request by sadhansood**
When reading downloaded checkpoint for events, we ensure that it is the right checkpoint summary (by verifying it is signed by the right committee and points to the previous verified checkpoint) but do not check checkpoint's content and events. This PR fixes this issue and adds the missing event verification.

### CODE REVIEW

File: crates/walrus-event/src/event_processor.rs [hunk 1]
```diff
@@ -154,6 +156,64 @@ impl EventProcessor {
...
+
+            if let Some((events_digest, events)) = transaction
+                .effects
+                .events_digest()
+                .zip(transaction.events.as_ref())
+            {
+                if *events_digest != events.digest() {
+                    bail!("Events digest does not match");
+                }
+            }
```
Comment (halfprice) [inline crates/walrus-event/src/event_processor.rs:285]: Is this part still necessary if the event digest match?

File: crates/walrus-event/src/event_processor.rs [hunk 2]
```diff
@@ -154,6 +156,64 @@ impl EventProcessor {
...
+        )
+        .map_err(|checkpoint| {
+            anyhow!(
+                "Failed to verify sui checkpoint: {}",
+                checkpoint.sequence_number
+            )
+        })?;
+
+        // Verify that checkpoint summary matches the content
+        if verified_checkpoint.content_digest != *checkpoint.checkpoint_contents.digest() {
```
Comment (halfprice) [inline crates/walrus-event/src/event_processor.rs:255]: There is also a `verify_with_contents` in CertifiedCheckpointSummary

File: crates/walrus-event/src/event_processor.rs (hunk 1 above)
Comment (sadhansood) [reply crates/walrus-event/src/event_processor.rs:285]: Yes good question, once the effects digest match, we compare the events digest to ensure they match with the events in the `CheckpointData` struct.

File: crates/walrus-event/src/event_processor.rs (hunk 2 above)
Comment (sadhansood) [reply crates/walrus-event/src/event_processor.rs:255]: It is pretty much the same thing but we have `TrustedCheckpoint` in here whereas that function takes `CertifiedCheckpoint`, so a little bit of extra conversion would be required and hence this one liner makes sense

**Pull request merged**